homeassistant/components/wallbox/number.py

```diff
@@ -66,21 +67,30 @@ def __init__(
         entry: ConfigEntry,
         description: WallboxNumberEntityDescription,
     ) -> None:
-        """Initialize a Wallbox sensor."""
+        """Initialize a Wallbox number entity."""
         super().__init__(coordinator)
         self.entity_description = description
         self._coordinator = coordinator
         self._attr_name = f"{entry.title} {description.name}"
         self._attr_unique_id = f"{description.key}-{coordinator.data[CHARGER_DATA_KEY][CHARGER_SERIAL_NUMBER_KEY]}"
+        self._is_bidirectional = any(
+            coordinator.data[CHARGER_DATA_KEY][CHARGER_PART_NUMBER_KEY][0:3] in model
+            for model in BIDIRECTIONAL_MODEL_PREFIXES
+        )
```

Comment:
I'm not sure we need the loop here.
Does this not work?
```python
self._is_bidirectional = coordinator.data[CHARGER_DATA_KEY][CHARGER_PART_NUMBER_KEY][0:3] in BIDIRECTIONAL_MODEL_PREFIXES
```

Reply:
yes it does.. sorry not my native language and got a little carried away.  Have pushed the change you suggested.